compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -587,22 +587,22 @@ private void visitAssignment(BLangExpression varRefExpr, TaintedStatus varTainte
             if (varTaintedStatus == TaintedStatus.TAINTED && varRefExpr instanceof BLangVariableReference) {
                 BLangVariableReference varRef = (BLangVariableReference) varRefExpr;
                 if (isMutableVariable(varRef) && isGlobalVarOrServiceVar(varRef) && !isMarkedTainted(varRef)) {
```

Comment:
We seem to be repeating the `isStructuredAccessOnVariableReference()` check in `isMutableVariable()` and `isGlobalVarOrServiceVar()`. Can we reconsider the checks?

Reply:
I'm not quite sure if I'm getting your idea, what we want it to check the `expr` part of AccessExpressions

Reply:
What I meant was that we seem to be checking `isStructuredAccessOnVariableReference()` in both `isMutableVariable()` 
https://github.com/ballerina-platform/ballerina-lang/pull/19155/files#diff-25120e77c4c9017615d43d8e159cb3baR650
 and `isGlobalVarOrServiceVar()` https://github.com/ballerina-platform/ballerina-lang/pull/19155/files#diff-25120e77c4c9017615d43d8e159cb3baR684

Reply:
Got it, 
`isStructuredAccessOnVariableReference` is a check to decide weather to perform the logic in isMutableVariable and isGlobalVarOrServiceVar or to go one step in and perform the logic. I found it more readable to extract the checks into a function hence `isStructuredAccessOnVariableReference` was born.